src/main/java/com/bitdecay/game/system/RespawnSystem.java

```diff
@@ -43,6 +43,8 @@ protected void forEach(float delta, MyGameObject gob) {
                     gob.forEach(PhysicsComponent.class, phy -> {
                         phy.body().aabb.xy.x = resPos.x;
                         phy.body().aabb.xy.y = resPos.y;
+                        // TODO: this SHOULD be setting the velocity to 0... but it isn't?
+                        phy.body().velocity.scale(0);
```

Comment:
scale returns a new BitPoint (I'm going to go through and rename some of these to avoid this problem)